spyder/plugins/toolbar/container.py

```diff
@@ -317,7 +317,6 @@ def create_toolbars_menu(self):
         for toolbar_id, toolbar in self._ADDED_TOOLBARS.items():
             if toolbar:
                 action = toolbar.toggleViewAction()
-                action.__class__ = QActionID

```

Comment:
If this is not possible in PySide2, please skip it just for it so we can have it for PyQt.

I know it's a hack, but we really need it until we're able to remove it in some months.

Reply:
I removed this change (the pull request now doesn’t modify `container.py`).

Reply:
Ok, but how did you manage to solve the problem with PySide2?

Reply:
I did not. Changing `__class__` of a `QObject` created by C++ (e.g. `QMainWindow::addToolbar(QString)`, `QToolBar::addAction(QString)`, `QToolBar::toggleViewAction()`) seems to invalidate the corresponding Python (PySide) object. I have no idea how to work around this, if possible at all. (Changing `__class__` of `QObject` created in Python (e.g. `a = PySide2.QtWidgets.QAction()`) seems to work.)

Can you point to a PR or issue where `QActionID` is used? My question being: Why exactly is `QActionID` and changing of `__class__` required? Do you want to change the type of `QAction` to `QActionID` (because you you want `isinstance(ins, QActionID)` to be true)? Or do you use `QActionID` just to add the `action_id` attribute?

Reply:
This is the PR that added the change: https://github.com/spyder-ide/spyder/pull/16205

As I said, if this is not possible for PySide2 but things are working fine without it, you can add an `if not PYSIDE2` (qtpy has that constant) so it's not used in that case.

Reply:
> As I said, if this is not possible for PySide2 but things are working fine without it, you can add an if not PYSIDE2 (qtpy has that constant) so it's not used in that case.

Sorry, I misunderstood that this is what you want to do. I will shortly push this change.

Reply:
Done, see commit "pyside: Do not modify __class__ in toolbar plugin (prevents startup)"

Reply:
Thanks! That's exactly what I had in mind.